src/engine/GarbageCollector.ts

```diff
@@ -0,0 +1,82 @@
+export interface GarbageCollectionOptions {
+  textureCollectInterval: number; // default 60_000 ms
+  fontCollectInterval: number; // default 60_000 ms
+  textMeasurementCollectInterval: number; // default 60_000 ms
+}
+
+export interface GarbageCollectorOptions {
+  /**
+   * Returns a timestamp in milliseconds representing now
+   */
+  nowFn: () => number;
```

Comment:
```suggestion
  getTimestamp: () => number;
```
`nowFunction` felt a bit weird